unstructured/ingest/connector/github.py

```diff
@@ -52,23 +54,71 @@ class GitHubIngestDoc(GitIngestDoc):
     connector_config: SimpleGitHubConfig
     registry_name: str = "github"
 
-    def _fetch_and_write(self) -> None:
-        content_file = self.connector_config.get_repo().get_contents(self.path)
+    @property
+    def date_created(self) -> t.Optional[str]:
+        return None
+
+    @requires_dependencies(["github"], extras="github")
+    def _fetch_content(self, is_content_file=False):
+        from github.GithubException import UnknownObjectException
+
+        try:
+            content_file = self.connector_config.get_repo().get_contents(self.path)
+        except UnknownObjectException:
+            logger.error(f"File doesn't exists {self.connector_config.url}/{self.path}")
+            return None
+        except Exception:
+            logger.error(f"Error processing {self.connector_config.url}/{self.path}")
+            raise
```

Comment:
we're not really "processing" just fetching, but also ..why do this? why not just let it raise the exception instead of catch + reraise?